### DISCUSSION TITLE
Using onOpenAutoFocus for state setters in a Dialog

### DISCUSSION THREAD

**Discussion by RheingoldRiver** [radix-ui/primitives#2467] [Help]
Hi,
I'm using the Dialog component to make a settings modal. I don't want the settings to be "legacy," so they have their internal state that mirrors several global app state variables. I want to reinitialize these every time the settings menu is reopened. The only way I can see to do this is to use `onOpenAutoFocus` and put a setter there, and then *not* call event.preventDefault().

Here is my onOpenAutoFocus function:

```jsx
      onOpenAutoFocus={() => {
        setCurrentState(getCurrentState()); // returns an object based on state from context
      }}
```

But, this doesn't seem like how the API is intended to be used, so I'm not sure if what I'm doing is right. Is there a better way to accomplish what I want?

**Comment by benoitgrelard** [comment]
Sorry i do not understand the issue/question.
